src/Widgets/Page.vala

```diff
@@ -108,13 +113,27 @@ namespace Network.Widgets {
                 try {
                     device.disconnect (null);
                 } catch (Error e) {
-                    warning (e.message);
+                    show_dialog (
+                        _("Failed To Disconnect"),
+                        _("Could not disconnect: %s.").printf (e.message),
+                        "dialog-error",
+                        Gtk.ButtonsType.CLOSE
+                    );
+
+                    control_switch.active = true;
                 }
             } else if (control_switch.active && device.get_state () == NM.DeviceState.DISCONNECTED) {
                 var connection = NM.SimpleConnection.new ();
                 var remote_array = device.get_available_connections ();
                 if (remote_array == null) {
-                    this.show_error ();
+                    show_dialog (
+                        _("Failed To Connect"),
+                        _("Could not find any available connections for this device to establish connection."),
```

Comment:
This sentence is a bit awkward. Can you try to explain when this would appear? Maybe I can help reword it

Reply:
This is shown when there is no suitable connection found for the device to connect with, you see those connections in the "Edit connections" dialog. When there are 0 of these, there is no way to connect because there is no configuration (NM usually does setup one though).

Reply:
Hm then wouldn't this show an error any time you're in a new place? That doesn't sound like the desired behavior. If the user doesn't have any connections set up, then not connecting to anything seems like expected behavior